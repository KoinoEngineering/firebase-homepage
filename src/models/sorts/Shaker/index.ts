import Comparsion, { ComparsionParams } from "../Comparsion";
interface ShakerAdditionalProperty {
  cursor: number;
  start: number;
  last: number;
  isUp: boolean;
}

type ShakerParams = ComparsionParams & Partial<ShakerAdditionalProperty>;

export default class Shaker
    extends Comparsion
    implements ShakerAdditionalProperty
{
  public name = "シェーカーソート";
  public cursor: number;
  public start: number;
  public last: number;
  public isUp: boolean;
  constructor({
      items,
      compareCount,
      swapCount,
      cursor,
      start,
      last,
      isUp,
  }: ShakerParams) {
      super({ items, compareCount, swapCount });
      this.cursor = cursor || 0;
      this.start = start || 0;
      this.last = last || items.length - 1;
      this.isUp = isUp ?? true;
  }

  public compare = async () => {
      const { items, compareCount } = this;
      if (!this.ended()) {
          return new Shaker({
              ...this,
              items: items.map((item, i) => {
                  if (i === this.cursor || i === this.cursor + 1) {
                      return {
                          ...item,
                          comparing: true,
                      };
                  } else {
                      return item;
                  }
              }),
              compareCount: compareCount + 1,
          });
      }
      return this;
  };

  public swap = async () => {
      if (!this.ended()) {
          let { swapCount } = this;
          const { items, cursor, start, last, isUp } = this;
          // 比較中のものを取り出す
          const comparings = items.filter((item) => item.comparing);

          //   とりあえず2個じゃなかったら落とす
          if (comparings.length !== 2) {
              throw new Error("The item being compared must be 2 items");
          }

          // 比較中のものの順序をそろえる
          const swapped =
        comparings[0].value > comparings[1].value
            ? (++swapCount, comparings.reverse())
            : comparings;

          let mapCursor = 0;
          const nextUp = cursor === (isUp ? last - 1 : start) ? !isUp : isUp;
          return new Shaker({
              ...this,
              items: items.map((item, i) => {
                  if (item.comparing) {
                      return {
                          ...swapped[mapCursor++],
                          comparing: false,
                          fixed: i === (isUp ? last : start) || start + 1 === last,
                      };
                  } else {
                      return item;
                  }
              }),
              isUp: nextUp,
              last: isUp && cursor === last - 1 ? last - 1 : last,
              start: !isUp && cursor === start ? start + 1 : start,
              cursor: this.getCursor(isUp, cursor, start, last),
              swapCount,
          });
      }
      return this;
  };
  private getCursor(
      isUp: boolean,
      cursor: number,
      start: number,
      last: number,
  ) {
      if (isUp) {
          if (cursor === last - 1) {
              return cursor - 1;
          } else {
              return cursor + 1;
          }
      } else {
          if (cursor === start) {
              return cursor + 1;
          } else {
              return cursor - 1;
          }
      }
  }
}
