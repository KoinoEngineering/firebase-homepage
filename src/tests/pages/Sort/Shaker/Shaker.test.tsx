import React from "react";
import Shaker from "src/pages/Sorts/Shaker/Shaker";
import { renderWithProvider } from "src/tests/tetutils";

test("Shaker expect to match snapshot", () => {
    const target = renderWithProvider(<Shaker />);
    expect(target).toMatchSnapshot();
});
