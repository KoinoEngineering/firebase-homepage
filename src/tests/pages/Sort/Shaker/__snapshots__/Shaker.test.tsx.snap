// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shaker expect to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root ShakerPageContainer-root-3 MuiContainer-maxWidthLg"
        id="Shaker"
      >
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-6 MuiGrid-spacing-xs-2"
          id="TitleArea"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              シェーカーソート
            </h2>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-true"
          >
            <a
              href="/sorts"
            >
              戻る
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-8 MuiGrid-spacing-xs-4"
          id="SettingArea"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
                data-shrink="true"
              >
                要素数
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  disabled=""
                  type="text"
                  value="20"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                  >
                    <span>
                      要素数
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-root-13 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                向き
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <div
                  aria-haspopup="listbox"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  role="button"
                  tabindex="0"
                >
                  昇順
                </div>
                <input
                  type="hidden"
                  value="asc"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                  >
                    <span>
                      向き
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
                data-shrink="true"
              >
                処理間隔
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  disabled=""
                  type="text"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                  >
                    <span>
                      処理間隔
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-15"
          id="ControlArea"
        >
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-17"
            id="Reset"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3"
            >
              リセット
            </h3>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-19 MuiGrid-spacing-xs-4"
            id="Reset"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  昇順
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  降順
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  ランダム
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-21"
            id="ControlArea"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  開始
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-23"
          id="ContentsArea"
        >
          <div
            class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
            id="SquareContainer"
          >
            <div
              class="MuiGrid-root makeStyles-root-24 MuiGrid-container MuiGrid-align-items-xs-flex-end"
              id="ReplaceSortContents"
            >
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-27"
              >
                100
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-29"
              >
                95
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-31"
              >
                90
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-33"
              >
                85
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-35"
              >
                80
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-37"
              >
                75
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-39"
              >
                70
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-41"
              >
                65
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-43"
              >
                60
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-45"
              >
                55
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-47"
              >
                50
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-49"
              >
                45
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-51"
              >
                40
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-53"
              >
                35
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-55"
              >
                30
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-57"
              >
                25
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-59"
              >
                20
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-61"
              >
                15
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-63"
              >
                10
              </div>
              <div
                class="MuiGrid-root makeStyles-root-25 makeStyles-root-65"
              >
                5
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container"
            id="ChartContainer"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 200px;"
            >
              <div
                style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root ShakerPageContainer-root-3 MuiContainer-maxWidthLg"
      id="Shaker"
    >
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-6 MuiGrid-spacing-xs-2"
        id="TitleArea"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2"
          >
            シェーカーソート
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-true"
        >
          <a
            href="/sorts"
          >
            戻る
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-8 MuiGrid-spacing-xs-4"
        id="SettingArea"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
              data-shrink="true"
            >
              要素数
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled=""
                type="text"
                value="20"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                >
                  <span>
                    要素数
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-root-13 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              向き
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                role="button"
                tabindex="0"
              >
                昇順
              </div>
              <input
                type="hidden"
                value="asc"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                >
                  <span>
                    向き
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
              data-shrink="true"
            >
              処理間隔
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled=""
                type="text"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                >
                  <span>
                    処理間隔
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-15"
        id="ControlArea"
      >
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-17"
          id="Reset"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            リセット
          </h3>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-19 MuiGrid-spacing-xs-4"
          id="Reset"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                昇順
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                降順
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                ランダム
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-21"
          id="ControlArea"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                開始
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-4 GridRow-container-23"
        id="ContentsArea"
      >
        <div
          class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
          id="SquareContainer"
        >
          <div
            class="MuiGrid-root makeStyles-root-24 MuiGrid-container MuiGrid-align-items-xs-flex-end"
            id="ReplaceSortContents"
          >
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-27"
            >
              100
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-29"
            >
              95
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-31"
            >
              90
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-33"
            >
              85
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-35"
            >
              80
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-37"
            >
              75
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-39"
            >
              70
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-41"
            >
              65
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-43"
            >
              60
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-45"
            >
              55
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-47"
            >
              50
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-49"
            >
              45
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-51"
            >
              40
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-53"
            >
              35
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-55"
            >
              30
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-57"
            >
              25
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-59"
            >
              20
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-61"
            >
              15
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-63"
            >
              10
            </div>
            <div
              class="MuiGrid-root makeStyles-root-25 makeStyles-root-65"
            >
              5
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
          id="ChartContainer"
        >
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 200px;"
          >
            <div
              style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
